### PR TITLE
Prepare reqres for streaming

### PR DESCRIPTION
**Pull request by jcorbin**
Final code rotation around reqres to prepare for streaming.

Reviewers: @kriskowal @Raynos

### CODE REVIEW

File: node/reqres.js [hunk 1]
```diff
@@ -47,13 +48,29 @@ function TChannelIncomingRequest(id, options) {
     self.headers = options.headers || {};
     self.checksum = options.checksum || null;
     self.checksumType = options.checksumType || 0;
-    self.arg1 = options.arg1 || emptyBuffer;
```
Comment (Raynos) [inline node/reqres.js]: Any reason to not pre-define these as `null` in the constructor ?

File: node/reqres.js [hunk 2]
```diff
@@ -134,7 +205,13 @@ TChannelOutgoingRequest.prototype.hookupCallback = function hookupCallback(callb
     }
     function onResponse(res) {
         self.removeListener('error', onError);
-        callback(null, res);
+        if (callback.canStream) {
```
Comment (Raynos) [inline node/reqres.js]: I dont understand the difference between streaming and non streaming here.

Why would one need zalgo protection and not the other ?

**Comment by Raynos** [comment]
lgtm.

File: node/reqres.js (hunk 2 above)
Comment (jcorbin) [inline node/reqres.js]: it's a stub for later use, it shouldn't make a ton of sense yet

File: node/reqres.js (hunk 1 above)
Comment (jcorbin) [inline node/reqres.js]: Because they're going away real soon at this layer, to be provided by the upcoming use of argstream as base class in a future pr.

**Pull request merged**